game-core/src/main/java/games/strategy/engine/lobby/client/ui/LobbyFrame.java

```diff
@@ -52,7 +52,12 @@ public LobbyFrame(final LobbyClient client, final LobbyServerProperties lobbySer
     chatPlayers.setChat(chat);
     chatPlayers.setPreferredSize(new Dimension(200, 600));
     chatPlayers.addActionFactory(this::newAdminActions);
-    final LobbyGamePanel gamePanel = new LobbyGamePanel(this.client.getMessengers());
+
+    final LobbyGameTableModel tableModel =
+        new LobbyGameTableModel(client.listGames(), client.isAdmin());
+    client.addGameChangeListener(tableModel);
```

Comment:
Before LobbyGameTableModel had an inner object that was an instance of the listener class, the model class tooke as parameter messengers and then registered the inner listener with the messengers. This violates IOC principle, and secondly the inner listener class is not needed when the game table model can implement that interface directly.

Once the listener is not a private variable, we can set up the listener here, outside of the game table model constructor.